### ISSUE TITLE
Inconsistent identification of licenses in sources

### ISSUE THREAD

**Issue by MGlauer** [OpenEnergyPlatform/examples#55]
The current metadata version seems to have two ways to identify licenses:

1. "`<metadata.licenses[*].name>`"
2. "`<licence.title>` (`<license.name>`)"

I have the feeling all sources should just use `license.name` as an identifier in order to maintain machine readability

**Comment by Ludee** [comment]
There are licenses which are not included in the SPDX list.
e.g. **dl-de-by-2.0**
Eine weitere Liste ist hier: https://github.com/fraunhoferfokus/ogd-metadata/blob/master/lizenzen/deutschland.json

**Comment by Ludee** [comment]
from https://frictionlessdata.io/specs/data-package/

**licenses**
The license(s) under which the package is provided.

This property is not legally binding and does not guarantee the package is licensed under the terms defined in this property.

licenses MUST be an array. Each item in the array is a License. Each MUST be an object. The object MUST contain a name property and/or a path property. It MAY contain a title property.

**Comment by Ludee** [comment]
The "title" is very useful field for humans. It is not needed by the data package standard but still important.

**Comment by MGlauer** [comment]
Yes, `title` *and* `name` should be there , but as individual fields. 

Source licenses use a mixture of both:

`"license": "Creative Commons Zero v1.0 Universal (CC0-1.0)"`

Which is not good for machine readability and for standard conformance. My proposal was to use an atomic identifier, I.e. just `"CC0-1.0" `. A verbose title can still be optional.

On 4 July 2019 16:23:39 GMT+02:00, "Ludwig Hülk" <notifications@github.com> wrote:
>The "title" is very useful field for humans. It is not needed by the
>data package standard but still important. 
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/OpenEnergyPlatform/examples/issues/55#issuecomment-508500249

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by christian-rli** [comment]
Source licenses are indeed not machine readable. While complete machine readability would've been nice we expect that within sources a plethora of potential licenses might show up that cannot all be covered in spdx or similar lists. So we thought it best to treat source licenses as comments. Do you think it should be named differently to signify that quality?

**Comment by Ludee** [comment]
I suggest to rename the field to "**licenseId**" and not use the full name in sources.
This comes from the [SPDX file](https://github.com/spdx/license-list-data/blob/master/json/licenses.json#L1161). 
Then we need to adapt the description in the wiki:
> License of the source. Complete name and license id in brackets. Standard: SPDX License List

Then we have the two license ID in the string: sources/licenseID and licenses/name.
Does this make sense?

**Comment by christian-rli** [comment]
I'm not sure I fully unterstand your suggestion @Ludee . Are you suggesting to use the following:

```
"sources": [
    {"licenseID": "CDDL-1.0"}],
```
and
```
"licenses": [
    {"name": "CDDL-1.0"}],
```
?

I don't think this would make sense, because there would be two different keys expecting the same kind of input. Also, it wouldn't consider the problem of having source input of unknown licenses and it would get rid of our human readable license description in licenses.

To me it would make more sense to keep
```
"sources": [
    {"license": "comment style license description"}],
```
or to rename "license" into something like "license_comment" to signal that no machine readable string is required here.  Also, I think it would make sense to repurpose "name" in licenses to show a descriptive name (like "title" did before) and use licenseID as in the example you linked:
```
"licenses": [
    {"licenseID": "CDDL-1.0",
    "name": "Common Development and Distribution License 1.0",}],

```
What do you think?

**Comment by Ludee** [comment]
Exactly, this is what I was trying to say. But:
> "licenses": The object MUST contain a _name_ property and/or a _path_ property. It MAY contain a _title_ property.

An additional thought is to remove the license information and add a link to the "metadata of the source". That would prevent having double information. 
But the original idea was to include all licenses (and copyright attribution) in the metadata. If the metadata strings are connected (like an FK) we won't need that any more.
"sourceId" -> ["id"](https://github.com/OpenEnergyPlatform/examples/blob/v1.4_release/metadata/oep_metadata_template.json#L3)

`"sources": [
    {"title": "", "description": "", "path": "", "sourceId": ""}],`

**Comment by christian-rli** [comment]
I don't see the problem, to be honest. There _is_ a name property in licenses in the suggestion. Is it defined to mean something like licenseID? Then I'd see the conflict.

I'm not completely opposed removing and linking license information, but not a fan either. I think it might be metaOverkill and just sticking to a simple json-string is likely the simpler/cleaner option.

**Comment by christian-rli** [comment]
Addressed this issue in #58 . There is now the same list of licenses fields within sources as there is in "licenses" itself.
